core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooReader.java

```diff
@@ -184,7 +184,11 @@ protected <R> R retryLoopMutator(ZKFunctionMutator<R> zkf,
       try {
         return zkf.apply(getZooKeeper());
       } catch (KeeperException e) {
-        if (alwaysRetryCondition.test(e) || useRetryForTransient(retries, e)) {
+        if (alwaysRetryCondition.test(e)) {
+          retries.waitForNextAttempt(log,
+              "attempting to communicate with zookeeper after exception that requires retry");
```

Comment:
Would it make sense to add anything about that exception to the message?

Reply:
Like how long we are waiting each time?
Also can I presume that there is a cap as to the longest amount of period we will wait?

Reply:
> Would it make sense to add anything about that exception to the message?

I can do that, but since the retry condition is passed in as a parameter, it won't be clear what about the exception triggered the retry, without also injecting a parameter to customize the log message for the given condition... which I think would complicate the method's API too much.

> Like how long we are waiting each time? Also can I presume that there is a cap as to the longest amount of period we will wait?

The wait time is logged at debug by the Retry class already. We don't log the max wait time or the backoff or other configuration, but it's hard-coded in the ZooReader class. We could log it, but I'm not sure where the best place to do that is where it would be useful.

Reply:
> which I think would complicate the method's API too much.

Agreed. Wondering if appending the message from the exception would be useful, but was not sure.

Reply:
> Like how long we are waiting each time? Also can I presume that there is a cap as to the longest amount of period we will wait?

Can find the config for the retry here :

https://github.com/apache/accumulo/blob/cd7642988bac831292e988326aaa96e07938c641/core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooReader.java#L45


Looks like max wait time is 5 seconds.  Wondering if that should be higher, like 30 seconds to a minute.  If collisions keep happening, backing off more is the best thing to do.

Reply:
I added the exception message to the log in 023415ae75728f90dc56dd0f6f9b0c75a157a68a